src/NumericalAlgorithms/FiniteDifference/Reconstruct.hpp

```diff
@@ -0,0 +1,70 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <utility>  // for std::pair
+
+#include "Utilities/Gsl.hpp"
+
+/// \cond
+template <size_t Dim>
+class Direction;
+template <size_t Dim, typename T>
+class DirectionMap;
+template <size_t Dim>
+class Index;
+/// \endcond
+
+namespace fd {
+/*!
+ * \ingroup FiniteDifferenceGroup
+ * \brief Variable and flux vector splitting reconstruction schemes for finite
+ * difference methods.
+ *
+ * Implementations of reconstruction methods must call this function
+ * (`fd::reconstruction::detail::reconstruct`) to perform the reconstruction.
+ * This function performs the actual loops taking into account strides and ghost
+ * zones for the reconstruction method. The type of reconstruction done is
```

Comment:
the function is (currently) in a `detail` namespace so it can't be documented. I'm not sure yet if it should be moved out or not. I'll add a snippet for now